codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -236,54 +206,81 @@ public void dataWriteShouldSucceed() throws Exception {
 
     @Test
     public void dataFramesShouldMerge() throws Exception {
+        createStream(STREAM_ID, false);
         final ByteBuf data = dummyData().retain();
-        DefaultChannelPromise secondPromise = new DefaultChannelPromise(channel, ImmediateEventExecutor.INSTANCE);
-        encoder.writeData(ctx, STREAM_ID, data, 0, true, promise);
-        encoder.writeData(ctx, STREAM_ID, data, 0, true, secondPromise);
+
+        ChannelPromise promise1 = newPromise();
+        encoder.writeData(ctx, STREAM_ID, data, 0, true, promise1);
+        ChannelPromise promise2 = newPromise();
+        encoder.writeData(ctx, STREAM_ID, data, 0, true, promise2);
+
+        // Now merge the two payloads.
         List<FlowControlled> capturedWrites = payloadCaptor.getAllValues();
         FlowControlled mergedPayload = capturedWrites.get(0);
         mergedPayload.merge(ctx, capturedWrites.get(1));
-
         assertEquals(16, mergedPayload.size());
-        assertFalse(secondPromise.isSuccess());
+        assertFalse(promise1.isSuccess());
```

Comment:
eventually we expect it to be successful, but at this point if it is done (failed or successful) then the test is failed. So instead of `assertFalse(promise1.isSuccess());` -> `assertFalse(promise1.isDone());`